src/main/java/com/databricks/jdbc/api/impl/DatabricksConnection.java

```
@@ -414,7 +414,22 @@ public SQLXML createSQLXML() throws SQLException {
   @Override
   public boolean isValid(int timeout) throws SQLException {
     ValidationUtil.checkIfNonNegative(timeout, "timeout");
-    return !isClosed();
+    if (isClosed()) {
+      return false;
+    }
+    String enableSqlValidation = connectionContext.getEnableSQLValidationForIsValid();
+
+    if ("1".equals(enableSqlValidation)) {
+      try (Statement stmt = createStatement()) {
+        stmt.setQueryTimeout(timeout);
+        stmt.execute("SELECT VERSION()");
+        return true;
+      } catch (SQLException e) {
```

nit : let's catch all exceptions